File: samples/java-server/src/main/java/com/farandula/models/FlightSegment.java
package com.farandula.models;

import com.farandula.Helpers.DateParser;
import com.nearsoft.farandula.models.AirLeg;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Created by antoniohernandez on 5/10/17.
 */
public class FlightSegment {

    private Airport departureAirport;
    private int departureDate;
    private Airport arrivalAirport;
    private int arrivalDate;
    private long duration;

    private String airLineMarketingName;
    private String airLineOperationName;
    private String airplaneData;

    private List<String> cabinTypes;

    public Airport getDepartureAirport() {
        return departureAirport;
    }

    public FlightSegment setDepartureAirport(Airport departureAirport) {
        this.departureAirport = departureAirport;
        return this;
    }

    public int getDepartureDate() {
        return departureDate;
    }

    public FlightSegment setDepartureDate(int departureDate) {
        this.departureDate = departureDate;
        return this;
    }

    public FlightSegment setDepartureDate(LocalDateTime departureDate) {
        int timestamp = DateParser.dateToTimestampSeconds(departureDate);
        this.departureDate = timestamp;
        return this;
    }

    public Airport getArrivalAirport() {
        return arrivalAirport;
    }

    public FlightSegment setArrivalAirport(Airport arrivalAirport) {
        this.arrivalAirport = arrivalAirport;
        return this;
    }

    public int getArrivalDate() {
        return arrivalDate;
    }

    public FlightSegment setArrivalDate (int arrivalDate) {
        this.arrivalDate = arrivalDate;
        return this;
    }

    public FlightSegment setArrivalDate (LocalDateTime arrivalDate) {
        int timestamp = DateParser.dateToTimestampSeconds(arrivalDate);
        this.arrivalDate = timestamp;
        return this;
    }

    public long getDuration() {
        return duration;
    }

    public FlightSegment setDuration(long duration) {
        this.duration = duration;
        return this;
    }

    public String getAirLineMarketingName() {
        return airLineMarketingName;
    }

    public FlightSegment setAirLineMarketingName(String airLineName) {
        this.airLineMarketingName = airLineName;
        return this;
    }

    public String getAirplaneData() {
        return airplaneData;
    }

    public FlightSegment setAirplaneData(String airplaneData) {
        this.airplaneData = airplaneData;
        return this;
    }

    public String getAirLineOperationName() {
        return airLineOperationName;
    }

    public FlightSegment setAirLineOperationName(String airLineOperationName) {
        this.airLineOperationName = airLineOperationName;
        return this;
    }

    public List<String> getCabinTypes() {
        return cabinTypes;
    }

    public FlightSegment setCabinTypes(List<String> cabinTypes) {
        this.cabinTypes = cabinTypes;
        return this;
    }
}

File: samples/ruby-server/app/models/FlightSegment.rb
require "date"
require "helpers/DateParser"

class FlightSegment

	def initialize (departureAirport, departureDate, arrivalAirport, arrivalDate, duration, airLineMarketingName, airLineOperationName, airplaneData, cabinTypes)
		@departureAirport = departureAirport
    	@departureDate = departureDate 
    	@arrivalAirport = arrivalAirport
    	@arrivalDate = arrivalDate
    	@duration = duration

    	@airLineMarketingName = airLineMarketingName
    	@airLineOperationName = airLineOperationName
    	@airplaneData = airplaneData

    	@cabinTypes = cabinTypes
    end


    def departureAirport
	  @departureAirport
	end

	# setter
	def departureAirport = (value)
	  @departureAirport = value
	end

	def departureDate
	  @departureDate
	end

	#TODO: Value of date / 1000
	def departureDate = (value)
	  @departureDate = DateTime.dateToTimeStampSeconds(value)
	end

	def arrivalAirport
	  @arrivalAirport
	end

	def arrivalAirport = (value)
	  @arrivalAirport = value
	end

	def arrivalDate
	  @arrivalDate
	end

	#TODO: Value of date / 1000
	def arrivalDate = (value)
	  @arrivalDate = DateTime.dateToTimeStampSeconds(value)
	end

	def duration
	  @duration
	end

	def duration = (value)
	  @duration = value
	end

	def airLineMarketingName
	  @airLineMarketingName
	end

	def airLineMarketingName = (value)
	  @airLineMarketingName = value
	end

	def airLineOperationName
	  @airLineOperationName
	end

	def airLineOperationName = (value)
	  @airLineOperationName = value
	end

	def airplaneData
	  @airplaneData
	end

	def airplaneData = (value)
	  @airplaneData = value
	end

	def cabinTypes
	  @cabinTypes
	end

	def cabinTypes = (value)
	  @cabinTypes = value
	end


end



